arch/xtensa/src/common/xtensa_context.S

```diff
@@ -164,16 +161,28 @@ _xtensa_context_save:
 #error Overlay support is not implemented
 #endif
 
-	s32i	a0, a2, (4 * REG_TMP0)			/* Save return address */
-	s32i	sp, a2, (4 * REG_TMP1)			/* Save current stack pointer */
-	wsr		a2, EXCSAVE_1					/* Preserve register save area */
-
-	l32i	sp, a2, (4 * REG_A1)			/* Restore the interruptee's SP */
-	call0	_xtensa_window_spill			/* Preserves only a4-a5, a8-a9, a12-a13 */
+    s32i  a0, sp, (4 * REG_TMP0)
+    rsr     a2, PS                     /* to be restored after SPILL_ALL_WINDOWS */
+    movi    a0, PS_INTLEVEL_MASK
+    and     a3, a2, a0                 /* get the current INTLEVEL */
+    bgeui   a3, XCHAL_EXCM_LEVEL, 1f   /* calculate max(INTLEVEL, XCHAL_EXCM_LEVEL) */
```

Comment:
I see that the third argument for `bgeui` is the label that will be the branch target, but I would expect it to be just `1`.
What does the `f` in `1f` mean? Or it it a typo?

Reply:
`f` means `forward` (and `b` means `backwards`). It's just a way in assembly to limit the number of labels.  Without it we would have to add unique labels for all jumps.

Reply:
Nice, didn't know about this one. Thanks for the clarification!